packages/@webex/internal-plugin-metrics/test/unit/spec/call-diagnostic/call-diagnostic-metrics-batcher.ts

```diff
@@ -0,0 +1,207 @@
+/*!
+ * Copyright (c) 2015-2020 Cisco Systems, Inc. See LICENSE file.
+ */
+
+import {assert} from '@webex/test-helper-chai';
+import FakeTimers from '@sinonjs/fake-timers';
+import Metrics, {config} from '@webex/internal-plugin-metrics';
+import MockWebex from '@webex/test-helper-mock-webex';
+import sinon from 'sinon';
+import {WebexHttpError} from '@webex/webex-core';
+
+function promiseTick(count) {
+  let promise = Promise.resolve();
+
+  while (count > 1) {
+    promise = promise.then(() => promiseTick(1));
+    count -= 1;
+  }
+
+  return promise;
+}
+
+describe('plugin-metrics', () => {
+  describe('CallDiagnosticEventsBatcher', () => {
+    let webex;
+
+    beforeEach(() => {
+      //@ts-ignore
+      webex = new MockWebex({
+        children: {
+          metrics: Metrics,
+        },
+      });
+
+      webex.config.metrics = config.metrics;
+
+      webex.request = function (options) {
+        return Promise.resolve({
+          statusCode: 204,
+          body: undefined,
+          options,
+        });
+      };
+      sinon.spy(webex, 'request');
+    });
+
+    let clock;
+
+    beforeEach(() => {
+      //@ts-ignore
+      clock = FakeTimers.install();
+    });
+
+    afterEach(() => {
+      clock.uninstall();
+    });
+
+    describe('#request()', () => {
+      describe('when the request completes successfully', () => {
+        it('clears the queue', () => {
+          clock.uninstall();
+
+          return webex.internal.metrics.callDiagnosticEventsBatcher
+            .request({
+              type: 'diagnostic-event',
+              eventPayload: {
+                originTime: {
+                  triggered: 'mock triggered timestamp',
+                },
+              },
+            })
+            .then(() => {
+              //@ts-ignore
+              assert.calledOnce(webex.request);
+              assert.lengthOf(webex.internal.metrics.callDiagnosticEventsBatcher.queue, 0);
+            });
+        });
+      });
+
+      describe('when the request fails due to network disconnect', () => {
+        it('reenqueues the payload', () => {
+          // sinon appears to have gap in its api where stub.onCall(n) doesn't
```

Comment:
have you tried `stub.onCall(n).callsFake((options) => ....)`?

Reply:
see previous comment, I don't want to refactor the existing tests.